From: Jason Abele <jason.abele@gmail.com>
Date: Thu, 12 Nov 2015 16:04:28 -0800
Subject: rtl8723bs_mp_driver: Fix failure to start transmit

In MP mode, sometimes the transmissions will fail to start when
requested.  Realtek support recommended this fix.
---
 os_dep/linux/ioctl_linux.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/os_dep/linux/ioctl_linux.c b/os_dep/linux/ioctl_linux.c
index cce7250..6ae1f99 100644
--- a/os_dep/linux/ioctl_linux.c
+++ b/os_dep/linux/ioctl_linux.c
@@ -11743,6 +11743,11 @@ static int rtw_mp_ctx(struct net_device *dev,
 			return -EFAULT;
 			
 	DBG_871X("%s: in=%s\n", __func__, extra);
+	if (padapter->adapter_type == SECONDARY_ADAPTER) {
+		sprintf( extra, "Error: MP mode can't support Virtual Adapter, Please to use main Adapter\n");
+		wrqu->length = strlen(extra);
+		return 0;
+	}
 
 	countPkTx = strncmp(extra, "count=", 5); // strncmp TRUE is 0
 	cotuTx = strncmp(extra, "background", 20);
@@ -11784,8 +11789,10 @@ static int rtw_mp_ctx(struct net_device *dev,
 		bStartTest = 0; // To set Stop
 		pmp_priv->tx.stop = 1;
 		sprintf( extra, "Stop continuous Tx");
+		rtw_write8(padapter, 0xc50, 0x20);
 	} else {
 		bStartTest = 1;
+		rtw_write8(padapter, 0xc50, 0x7f);
 		if (pmp_priv->mode != MP_ON) {
 			if (pmp_priv->tx.stop != 1) {
 				DBG_871X("%s: MP_MODE != ON %d\n", __func__, pmp_priv->mode);
@@ -11956,7 +11963,11 @@ static int rtw_mp_arx(struct net_device *dev,
 			return -EFAULT;
 
 	DBG_871X("%s: %s\n", __func__, input);
-
+	if (padapter->adapter_type == SECONDARY_ADAPTER) {
+		sprintf( extra, "Error: MP mode can't support Virtual Adapter, Please to use main Adapter\n");
+		wrqu->length = strlen(extra);
+		return 0;
+	}
 	bStartRx = (strncmp(input, "start", 5)==0)?1:0; // strncmp TRUE is 0
 	bStopRx = (strncmp(input, "stop", 5)==0)?1:0; // strncmp TRUE is 0
 	bQueryPhy = (strncmp(input, "phy", 3)==0)?1:0; // strncmp TRUE is 0
